cogs/welcome/welcome.py

```diff
@@ -42,15 +50,63 @@ def __init__(self, bot: Red):
         self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
         self.config.register_guild(**DEFAULT_GUILD)
 
+    async def getRandomMessage(self, guild):
+        greetings = await self.config.guild(guild).KEY_GREETINGS()
+        numGreetings = len(greetings)
+        if numGreetings == 0:
+            return "Welcome to the server {USER}"
+        else:
+            return greetings[random.randint(0, numGreetings - 1)][1]
+
     # The async function that is triggered on new member join.
     @commands.Cog.listener()
     async def on_member_join(self, newMember: discord.Member):
-        await self.sendWelcomeMessage(newMember)
+        guild = newMember.guild
+        await self.sendWelcomeMessageChannel(newMember, guild)
+        # await self.sendWelcomeMessage(newMember)
 
     @commands.Cog.listener()
     async def on_member_remove(self, leaveMember: discord.Member):
         await self.logServerLeave(leaveMember)
 
+    # This async function is to look for if the welcome channel was removed
+    @commands.Cog.listener()
+    async def on_channel_remove(self, removedChannel: discord.TextChannel):
+        guild = removedChannel.guild
+        # the channel to post welcome stuff in
+        welcomeIDSet = await self.config.guild(guild).KEY_WELCOME_CHANNEL_SET()
+        welcomeID = await self.config.guild(guild).KEY_WELCOME_CHANNEL()
+        # the channel msg should point to
+        welcomeLinkIDSet = await self.config.guild(guild).KEY_WELCOME_CHANNEL_LINK_SET()
+        welcomeLinkID = await self.config.guild(guild).KEY_WELCOME_CHANNEL_LINK()
+        if welcomeIDSet and removedChannel.id == welcomeID:
+            await self.config.guild(guild).KEY_WELCOME_CHANNEL_SET.set(False)
+        return
+
+    async def sendWelcomeMessageChannel(self, newUser: discord.Member, guild: discord.guild):
+        channelID = await self.config.guild(guild).KEY_WELCOME_CHANNEL()
+        isSet = await self.config.guild(guild).KEY_WELCOME_CHANNEL_SET()
+        # if channel isn't set
+        if not isSet:
+            return
+        channel = discord.utils.get(guild.channels, id=channelID)
+        rawMessage = await self.getRandomMessage(guild)
+        splitMessage = rawMessage.split("{USER}")
+        message = ""
+        for x in range(len(splitMessage)):
+            message += splitMessage[x]
+            if x < len(splitMessage) - 1:
+                message += newUser.mention
+        fullMessage = f"""{message} Welcome to SFU Anime Club's Discord!
+
+Please check out <#225044755887685632> for all of our basic information and rules, and feel free to introduce yourself with the pinned template in this channel!
+
+We hope you enjoy your stay~
+
+=========================================================="""
```

Comment:
Can we not hardcode this? Also, why do we need this if we're creating a greeting?

Reply:
Removed hardcoding, sorry ono